Description: Move libdsocksd0.so to /usr/lib/dante-client/
Forwarded: not-needed
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2010-07-12

--- a/bin/socksify.in
+++ b/bin/socksify.in
@@ -49,7 +49,7 @@
 
 SOCKSIFY_PRELOAD_LIBS="@SOCKSIFY_PRELOAD_LIBS@"
 
-LIBDIR="@LIBRARY_PREFIX@"
+LIBDIR="@LIBRARY_PREFIX@/dante-client"
 
 #dlib/Makefile.am libtool flags should produce a predictable library name
 LIBRARY="${SOCKS_LIBRARY-${LIBDIR}/libdsocksd.@SOLIB_POSTFIX@}"
--- a/dlib/Makefile.in
+++ b/dlib/Makefile.in
@@ -206,7 +206,7 @@
 includedir = @includedir@
 infodir = @infodir@
 install_sh = @install_sh@
-libdir = @libdir@
+libdir = @libdir@/dante-client
 libexecdir = @libexecdir@
 localedir = @localedir@
 localstatedir = @localstatedir@
