Description: Teach dante-client to detect the executable's ABI.
Forwarded: not-needed
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2016-04-06

--- a/bin/socksify.in
+++ b/bin/socksify.in
@@ -70,26 +70,43 @@
     export DYLD_FORCE_FLAT_NAMESPACE=t
 fi
 
+MULTIARCH_FIND_LIB="/usr/lib/dante-client/multiarch-find-lib"
+
 #dlib/Makefile.am libtool flags should produce a predictable library name
 #(ending in SOLIB_POSTFIX).
-LIBRARY="${SOCKS_LIBRARY:-${FULLPATH}libdsocksd.@SOLIB_POSTFIX@}"
+if test x"$MULTIARCH_FIND_LIB" != x; then
+    LIBRARY="dante-client/libdsocksd.@SOLIB_POSTFIX@"
+    slibs=''
+    for lib in $SOCKSIFY_PRELOAD_LIBS; do
+    	base=`basename "$lib"`
+	slibs="$slibs $base"
+    done
+    SOCKSIFY_PRELOAD_LIBS="${slibs# }"
+else
+    LIBRARY="${SOCKS_LIBRARY:-${FULLPATH}libdsocksd.@SOLIB_POSTFIX@}"
+fi
 
 #platform dependent LD_PRELOAD-variant format
 PRELOAD_SEPERATOR="@PRELOAD_SEPERATOR@"
 PRELOAD_POSTFIX="@PRELOAD_POSTFIX@"
 
 #set variable activating preloading (LD_PRELOAD etc.)
-@PRELOAD_VARIABLE@="${LIBRARY}${SOCKSIFY_PRELOAD_LIBS:+${PRELOAD_SEPERATOR}}${SOCKSIFY_PRELOAD_LIBS}${PRELOAD_POSTFIX:+${PRELOAD_SEPERATOR}}${PRELOAD_POSTFIX}"
-export @PRELOAD_VARIABLE@
-
-LD_LIBRARY_PATH="${SOCKS_LIBDIR}${LD_LIBRARY_PATH:+:}${LD_LIBRARY_PATH}"
-export LD_LIBRARY_PATH
-
-#SunOS 64-bit library path
-if test x"@ISA64DIR@" != x; then
-    SOCKS_LIB64DIR="${LIB64DIR:-${SOCKS_LIBDIR}/@ISA64DIR@}"
-    LD_LIBRARY_PATH_64="${SOCKS_LIB64DIR}${LD_LIBRARY_PATH_64:+:}${LD_LIBRARY_PATH_64}"
-    export LD_LIBRARY_PATH_64
+if test x"$MULTIARCH_FIND_LIB" != x; then
+    @PRELOAD_VARIABLE@=`"$MULTIARCH_FIND_LIB" -e "$1" "$LIBRARY" $SOCKSIFY_PRELOAD_LIBS`
+    export @PRELOAD_VARIABLE@
+else
+    @PRELOAD_VARIABLE@="${LIBRARY}${SOCKSIFY_PRELOAD_LIBS:+${PRELOAD_SEPERATOR}}${SOCKSIFY_PRELOAD_LIBS}${PRELOAD_POSTFIX:+${PRELOAD_SEPERATOR}}${PRELOAD_POSTFIX}"
+    export @PRELOAD_VARIABLE@
+
+    LD_LIBRARY_PATH="${SOCKS_LIBDIR}${LD_LIBRARY_PATH:+:}${LD_LIBRARY_PATH}"
+    export LD_LIBRARY_PATH
+
+    #SunOS 64-bit library path
+    if test x"@ISA64DIR@" != x; then
+        SOCKS_LIB64DIR="${LIB64DIR:-${SOCKS_LIBDIR}/@ISA64DIR@}"
+        LD_LIBRARY_PATH_64="${SOCKS_LIB64DIR}${LD_LIBRARY_PATH_64:+:}${LD_LIBRARY_PATH_64}"
+        export LD_LIBRARY_PATH_64
+    fi
 fi
 
 exec "$@"
