Description: Fix some typographical and grammatical errors.
Forwarded: not-yet
Author: Peter Pentchev <roam@ringlet.net>
Last-Update: 2016-04-06

--- a/INSTALL
+++ b/INSTALL
@@ -251,7 +251,7 @@
 `--disable-preload'
      Do not build libdsocks.  The dynamic socks library might not work
      properly on all architectures.  This option disables building of
-     the library. Also disables it's use in the server.
+     the library. Also disables its use in the server.
 
 `--disable-clientdl'
      Disable building of libdsocks, used by the socksify application.
@@ -381,7 +381,7 @@
 
 No configuration-files are installed, these must be installed
 manually.
-The Dante server expects it's configuration file to be in /etc/danted.conf.
+The Dante server expects its configuration file to be in /etc/danted.conf.
 The client library uses the file /etc/dante.conf.
 
 The file bin/socksify.sh contains a shell script that allows one
--- a/NEWS
+++ b/NEWS
@@ -943,7 +943,7 @@
    instead of LOG_ERR, by popular request.
 
  o New module available: redirect, gives control over both
-   where clients requests and replies will end up, aswell as what
+   where clients requests and replies will end up, as well as what
    addresses and portranges the Dante server will use.
    Can also be used to limit the number of concurrent sessions
    from each client.
@@ -968,7 +968,7 @@
    default value set to "none".
    The global "method" is only used for socks-rules now.
 
- o delay checking of password/etc til we have received the socks
+ o delay checking of password/etc till we have received the socks
    request (rather than during negotiation, as was the case).
    Required for supporting passwordbased authentication via non-socks
    methods, e.g. pam.
@@ -1155,7 +1155,7 @@
        The correct usage should be that the method field is relative
        to the sourceaddress, and the sourceaddress for replies is
        a "non-socks" connection.  Special care might have to be
-       taken to not accidently block the replies since most of them
+       taken to not accidentally block the replies since most of them
        can only be using method "none" (the only other alternative
        is "rfc931").
 
--- a/doc/README.survey
+++ b/doc/README.survey
@@ -1,6 +1,6 @@
 	Dante development survey
 
 If you use this software, please take the time to fill out the survey
-at http://www.inet.no/dante/survey.html. It's purpose is to try to
+at http://www.inet.no/dante/survey.html. Its purpose is to try to
 determine which features are most wanted, and will probably influence
 the direction future development will take.
--- a/doc/sockd.conf.5
+++ b/doc/sockd.conf.5
@@ -1068,7 +1068,7 @@
 .IP
 If the route specifies that a username/password-method should be offered
 to the upstream proxy, \fBDante\fP will forward the username/password received
-from it's own client to the foreign upstream proxy, meaning the
+from its own client to the foreign upstream proxy, meaning the
 upstream proxy will receive the user's username and password in cleartext
 from \fBDante.
 .IP
--- a/doc/socksify.1
+++ b/doc/socksify.1
@@ -50,7 +50,7 @@
 to wrap all networking-related system calls.
 
 When used the script can socksify a program by simply prefixing
-the given program and it's arguments with the the socksify script,
+the given program and its arguments with the the socksify script,
 like in the example below.
 
 $ socksify ftp ftp.example.org
--- a/example/sockd.conf
+++ b/example/sockd.conf
@@ -118,12 +118,12 @@
 # Misc options.
 #
 
-# how many seconds can pass from when a client connects til it has
-# sent us it's request?  Adjust according to your network performance
+# how many seconds can pass from when a client connects till it has
+# sent us its request?  Adjust according to your network performance
 # and methods supported.
 #timeout.negotiate: 30   # on a lan, this should be enough.
 
-# how many seconds can the client and it's peer idle without sending
+# how many seconds can the client and its peer idle without sending
 # any data before we dump it?  Unless you disable tcp keep-alive for
 # some reason, it's probably best to set this to 0, which is
 # "forever".
--- a/include/common.h
+++ b/include/common.h
@@ -115,7 +115,7 @@
 #if PRERELEASE
 
 /*
- * Solaris 2.5.1 and it's stream stuff is broken and puts the processes
+ * Solaris 2.5.1 and its stream stuff is broken and puts the processes
  * into never-never land forever on half the sendmsg() calls if they
  * involve ancillary data.  (it seems to deadlock the processes.)
  * XXX need to retest what the current status of this is.
@@ -2188,7 +2188,7 @@
  * to getr the ipv4-mapped addresses returned to is to set ai_family to
  * zero, but then we get the regular ipv6-addresses also.  Since there
  * are cases when we want the ipv4-mapped addresses returned also, we
- * need to make the size set here able to accomodate that too.
+ * need to make the size set here able to accommodate that too.
  */
 #define MAX_ADDRINFO_NEXT (10)
 
@@ -3076,7 +3076,7 @@
 /*
  * Logs message "fmt" at priority "priority" to previously configured
  * output device.
- * Checks settings and ignores message if it's of to low a priority.
+ * Checks settings and ignores message if it's of too low a priority.
  */
 
 void vslog(int priority, const char *fmt, va_list ap, va_list apcopy)
@@ -3183,7 +3183,7 @@
  *
  * The route used may take into account the contents of "packet->req",
  * which is assumed to be the packet that will be sent to a socks server,
- * so it is recommended that it's contents be as conservative as possible.
+ * so it is recommended that its contents be as conservative as possible.
  *
  * When it has successfully connected to a gateway it will set
  * the packet->method members to point to the methods the gateway
@@ -3286,7 +3286,7 @@
  *
  * The route used may take into account the contents of "req", which is
  * assumed to be the packet that will be sent to a socks server, so it is
- * recommended that it's contents be as conservative as possible.
+ * recommended that its contents be as conservative as possible.
  *
  * Returns:
  *      On success: pointer to route that should be used.
@@ -4086,7 +4086,7 @@
 
 
 /*
- * Makes a dummy filedescriptor and returns it's index, or -1 on failure.
+ * Makes a dummy filedescriptor and returns its index, or -1 on failure.
  */
 
 #if DEBUG
--- a/include/monitor.h
+++ b/include/monitor.h
@@ -141,7 +141,7 @@
  * possibly progressing to the next state.
  *
  * Upon return, "mtu" will be updated to indicate its current state,
- * possibly after reseting the socket "s" to it's initial state and
+ * possibly after reseting the socket "s" to its initial state and
  * disabling further testing on this socket.
  */
 
--- a/include/sockd.h
+++ b/include/sockd.h
@@ -97,23 +97,23 @@
 
 /*
  * This is to handle a potential resource issue that can occur
- * in TCP when side 'A' of the TCP session closes it's end, but
- * the other end, side 'B', does not close it's end.  In this
+ * in TCP when side 'A' of the TCP session closes its end, but
+ * the other end, side 'B', does not close its end.  In this
  * situation, TCP will be forced to keep state for the TCP session
- * until side B closes it's end, or "never" if side B never closes.
+ * until side B closes its end, or "never" if side B never closes.
  *
  * Some kernels have added kernel support for tuning this on a global
  * basis, but implementations may vary.
  *
  * If this value is set, it gives the number of seconds to wait
- * for B to close it's side.  Note that this may break the application
+ * for B to close its side.  Note that this may break the application
  * protocol, as there may be nothing wrong with B using a long time,
- * days even, to close it's end.  It may however produce an unfortunate
- * resource problem with both the Dante server and the kernels TCP having to
+ * days even, to close its end.  It may however produce an unfortunate
+ * resource problem with both the Dante server and the kernel's TCP having to
  * keep state for these sessions, which in 99.999% of the cases could
  * probably be closed as B will not send anything more.
  *
- * The default therefor is to not enable this "feature".
+ * The default therefore is to not enable this "feature".
  */
 #define SOCKD_FIN_WAIT_2_TIMEOUT  (0) /* Seconds.  Set to 0 to disable. */
 
@@ -837,7 +837,7 @@
 
 /*
  * a request child can currently handle a maximum of one client, so can't
- * be changed and is therefor #defined here rather than in config.h.
+ * be changed and is therefore #defined here rather than in config.h.
  */
 #define SOCKD_REQUESTMAX   1
 
@@ -2154,9 +2154,9 @@
     *      but not both).
     *    - two (packets forwarded to both ipv4 and ipv6 addresses).
     * This however assumes our external interface has both IPv4 and IPv6
-    * addresses.  Does it not, at most one target socket will be created.
+    * addresses.  If it does not, at most one target socket will be created.
     *
-    * In Barefoots case, we don't have a TCP control-session, so the scenario
+    * In Barefoot's case, we don't have a TCP control-session, so the scenario
     * is one-to-many; we receive all udp packets from different clients on
     * the one source socket, and we forward them to various targets based
     * what client it was received from.  Since the target is hardcoded in
@@ -2284,7 +2284,7 @@
    /*
     * if not zero, this is an "old" client that has been sent back
     * to the negotiate process from the i/o process, due to the client
-    * changing it's target (remote http server).
+    * changing its target (remote http server).
     * "clientdata" contains the request received from the client,
     * already parsed into "request".
     */
@@ -2404,7 +2404,7 @@
                                      * should it be included in any counts.
                                      *
                                      * Basically just waiting for the SIGCHLD
-                                     * so we can add it's resourceusage to
+                                     * so we can add its resourceusage to
                                      * our counters.
                                      */
    unsigned char    exitingnormally;/* exiting normally, on our request?      */
@@ -2412,7 +2412,7 @@
    int              ack;            /* connection to child for acks.          */
    int              s;              /* connection to child for data.          */
 
-   pid_t            pid;            /* childs pid.                            */
+   pid_t            pid;            /* child's pid.                           */
    int              type;           /* child type.                            */
 
    time_t           created;        /* time created.                          */
@@ -2871,9 +2871,9 @@
  *
  * If "isnormalexit" is set, we are closing the pipes to this child and
  * expect it to exit normally.  This notifies the child about the close,
- * telling it to exit normally when done serving it's clients, unlike what
+ * telling it to exit normally when done serving its clients, unlike what
  * happens if the child exiting by itself without us telling it to do so,
- * or closing it's pipes to us first.
+ * or closing its pipes to us first.
  *
  * If "childpid" is 0, closes all children.
  */
@@ -3237,7 +3237,7 @@
  * as necessary (i.e., it's a "deep copy").
  *
  * Note that memory that is only set once at startup and never changed
- * is not copied from "old".  Instead it is copied from the processes
+ * is not copied from "old".  Instead it is copied from the process's
  * current config object.
  *
  * Returns 0 on success, -1 on failure.
@@ -4412,8 +4412,8 @@
 /*
  * Gets the outgoing IP address to use.
  *
- * "client_laddr" is address we accepted the client on.
- " "client_raddr" is the address of the client, on whos behalf we are
+ * "client_laddr" is the address we accepted the client on.
+ " "client_raddr" is the address of the client, on whose behalf we are
  *  binding an address on the external side.
  *
  * "command" is the SOCKS command the client requested.
@@ -4547,7 +4547,7 @@
 void
 io_handlesighup(void);
 /*
- * Called at sighup to let the i/o childs do what they need to do
+ * Called at sighup to let the i/o children do what they need to do
  * upon receiving a sighup.
  */
 
--- a/include/socks.h
+++ b/include/socks.h
@@ -610,7 +610,7 @@
  * If "takelock" is true, it means the function should take the
  * socksfdv/addrlock.
  *
- * The function duplicates all arguments in it's own form and does
+ * The function duplicates all arguments in its own form and does
  * not access the memory referenced by them afterwards.
  *
  * The function checks the state of all file descriptors on each call and
--- a/lib/Raccept.c
+++ b/lib/Raccept.c
@@ -437,7 +437,7 @@
    }
 
    /*
-    * a separate socket with it's own remote address and possibly different
+    * a separate socket with its own remote address and possibly different
     * local address too, so need to add it to the socksfd table.
     */
 
--- a/lib/Rconnect.c
+++ b/lib/Rconnect.c
@@ -122,14 +122,14 @@
                 * address and is now trying to connect out from it.
                 * That also indicates the socks server is listening on a port
                 * for this client.
-                * Can't accept() on a connected socket so lets close the
+                * Can't accept() on a connected socket so let's close the
                 * connection to the server so it can stop listening on our
                 * behalf, and we continue as if this was an ordinary connect().
-                * Can only hope the server will use same port as we for
+                * Can only hope the server will use same port as we do for
                 * connecting out.
                 *
                 * Client might get problems if it has done a getsockname(2)
-                * already, and thus thinks it knows it's local address,
+                * already, and thus thinks it knows its local address,
                 * as this Rconnect() will have to change it.
                 */
                int tmp_s;
@@ -164,7 +164,7 @@
                 * connect out on the same socket.  In this case
                 * we want to keep the port bound on the server, and
                 * just add a connect to the peer, so let udpsetup() do
-                * it's thing.
+                * its thing.
                 */
             }
             else
--- a/lib/address.c
+++ b/lib/address.c
@@ -822,7 +822,7 @@
     * With the OpenBSD thread implementation, if a thread is interrupted,
     * calling pthread_mutex_lock() seems to clear the interrupt flag, so
     * that e.g. select(2) will restart rather than returning EINTR.
-    * We don't wont that to happen since we depend on select(2)/etc.
+    * We don't want that to happen since we depend on select(2)/etc.
     * being interrupted by the process used to handle non-blocking connects.
     * We instead take the risk of not taking the thread-lock in this case.
     */
--- a/lib/clientprotocol.c
+++ b/lib/clientprotocol.c
@@ -1243,7 +1243,7 @@
                                            * RFC 1961 says GSS_C_DELEG_FLAG
                                            * should also be set, but I can't
                                            * see any reason why the client
-                                           * should want to forward it's
+                                           * should want to forward its
                                            * tickets to a socks server ...
                                            *
                                            * Don't set unless until we find
--- a/lib/gssapi.c
+++ b/lib/gssapi.c
@@ -890,7 +890,7 @@
     * Two modes: Buffered and unbuffered.
     *
     * Unbuffered:
-    *    We try to write upto "len" bytes, and if that fails, we
+    *    We try to write up to "len" bytes, and if that fails, we
     *    store the remaining bytes in our internal iobuf, encoded.
     *    This makes us able to return either "len" or -1 (only if fatal
     *    error) to caller, so that caller understands we have accepted
--- a/lib/hostcache.c
+++ b/lib/hostcache.c
@@ -64,7 +64,7 @@
 static size_t
 hosthash(const char *name, const size_t size);
 /*
- * Calculates a hash value for "name" and returns it's value.
+ * Calculates a hash value for "name" and returns its value.
  * Size of hash table is given by "size".
 */
 
@@ -91,7 +91,7 @@
 static size_t
 addrhash(const struct sockaddr_storage *addr, const size_t size);
 /*
- * Calculates a hash value for the IP-address "addr" and returns it's value.
+ * Calculates a hash value for the IP-address "addr" and returns its value.
  * Size of hash table is given by "size".
 */
 
--- a/lib/socket.c
+++ b/lib/socket.c
@@ -1121,7 +1121,7 @@
    /*
     * Test is to set a flag on fd1, and see if the same flag then gets set on
     * fd2.  Note that this flag must be a flag we can set on a socket that
-    * failed during connect(2), or where the peer has closed it's side
+    * failed during connect(2), or where the peer has closed its side
     * of the session, and which will be shared between descriptors that are
     * dup(2)'s of each other.
     *
--- a/lib/upnp.c
+++ b/lib/upnp.c
@@ -332,8 +332,8 @@
           * Can only find out what the external ip address of the device is.
           *
           * We could fetch the address here, but if the client never intends
-          * to find out what it's local address is, that's a waste of time.
-          * Therefor postpone it to the Rgetsockname() call, if it ever
+          * to find out what its local address is, that's a waste of time.
+          * Therefore postpone it to the Rgetsockname() call, if it ever
           * comes, and just connect(2) to the target for now, without
           * attempting to retrieve any information from the IGD.
           *
@@ -466,8 +466,8 @@
 
       case SOCKS_BIND: {
          /*
-          * Need tell the device to create a port mapping, mapping an
-          * address on it's side to the address we have bound.
+          * Need to tell the device to create a port mapping, mapping an
+          * address on its side to the address we have bound.
           * Then we need to get the ip address the device is using
           * on the external side.
           */
@@ -539,7 +539,7 @@
                                    (void *)GET_SOCKADDRADDR(&extaddr),
                                    NULL)) != 1) {
             snprintf(emsg, emsglen,
-                     "strange.  UPNP-device said it's external IP-address is "
+                     "strange.  UPNP-device said its external IP-address is "
                      "the %s \"%s\", but can't parse that with inet_pton(3).  "
                      "Errorcode %d (%s)",
                      safamily2string(extaddr.ss_family),
--- a/sockd/dante_udp.c
+++ b/sockd/dante_udp.c
@@ -1396,7 +1396,7 @@
 
    if (!ADDRISBOUND(expected)) {
       /*
-       * Client hasn't sent us it's complete address yet, but if
+       * Client hasn't sent us its complete address yet, but if
        * the parts of the address it has sent (if any) matches
        * the source of this packet, we have to assume this packet
        * is from it.  We can then update the expected address with
--- a/sockd/mother_util.c
+++ b/sockd/mother_util.c
@@ -63,8 +63,8 @@
 static void unexpecteddeath(void);
 /*
  * Should be called after any unexpected child death / child removal.
- * May disable creation of further children for a while and log a warning
- * if appropriate, or the enable creation of further children.
+ * May disable the creation of further children for a while and log a warning
+ * if appropriate, or enable the creation of further children.
  */
 
 void
@@ -1154,7 +1154,7 @@
              * Note that this might be a pid from our former self also
              * if we failed on an internal error, fork(2)-ed process to
              * get the coredump and continue.  When the fork(2)-ed process
-             * exits after generating the coredump, we will receive it's
+             * exits after generating the coredump, we will receive its
              * SIGCHLD, but no account of it.  To avoid that, hopefully
              * never happening, problem generating a recursive error, let
              * this be a swarnx(), and not a SWARNX().
--- a/sockd/rule.c
+++ b/sockd/rule.c
@@ -1535,15 +1535,15 @@
           * subsequent close(2) without us having sent the whole banner to
           * the client first.  But if the kernel wants to send RST while
           * we have data not yet sent, it will discard the data not yet
-          * sent.  We therefor drain the data first, trying to make sure the
+          * sent.  We therefore drain the data first, trying to make sure the
           * kernel does not discard the data in the outbuffer and sends a
           * RST when we close(2).  Note that this changes the RST to FIN.
           * See 2.17 in RFC 2525, "Failure to RST on close with data pending",
           * for more information about this.
           *
           * Note there is a race here, as the client could send us data
-          * between our last read(2) call and us closing the session later, not
-          * but much to do about that.
+          * between our last read(2) call and us closing the session later, but
+          * there's not much we can do about that.
           */
           char buf[1024];
           ssize_t p;
@@ -2458,7 +2458,7 @@
     * Set default values for some authentication-methods, if none
     * set.  Note that this needs to be set regardless of what the
     * method set in the rule is, as checkconfig() might add methods
-    * to the rules as part of it's operation.  This happens e.g. when
+    * to the rules as part of its operation.  This happens e.g. when
     * adding default methods to the global clientmethod line, if appropriate,
     * and then adding the same methods to the client-rules, if the rules
     * do not already have a method.
--- a/sockd/serverconfig.c
+++ b/sockd/serverconfig.c
@@ -63,7 +63,7 @@
  * Returns 0 if the address "ra" was addedd to the list of external addresses.
  *
  * Returns -1 if the address "ra" was not added for a non-fatal reason,
- * after loging a message if apropriate.
+ * after logging a message if appropriate.
  */
 
 static int addinternaladdr(const char *ifname,
@@ -73,7 +73,7 @@
  * Returns 0 if the address "ra" was addedd to the list of internal addresses.
  *
  * Returns -1 if the address "ra" was not added for a non-fatal reason,
- * after loging a message if apropriate.
+ * after logging a message if appropriate.
  */
 
 
@@ -741,7 +741,7 @@
    /* srchost, read from config file. */
    bzero(&config->srchost, sizeof(config->srchost));
 
-   /* stat: not touch.  Accumulated continously. */
+   /* stat: not touch.  Accumulated continuously. */
 
    /*
     * state; keep most of it, with the following exceptions:
--- a/sockd/shmem.c
+++ b/sockd/shmem.c
@@ -176,7 +176,7 @@
            function, SOCKD_SHMEMFILE);
 
    /*
-    * Hostcache also uses shmem, but has it's own area.
+    * Hostcache also uses shmem, but has its own area.
     */
    hostcachesetup();
 
--- a/sockd/sockd_child.c
+++ b/sockd/sockd_child.c
@@ -554,7 +554,7 @@
    size_t i, *childc;
 
 
-   slog(LOG_DEBUG, "%s: clearing all childs of type %s from set",
+   slog(LOG_DEBUG, "%s: clearing all children of type %s from set",
         function, childtype2string(childtype));
 
    setchildtype(childtype, &childv, &childc, NULL);
@@ -1116,7 +1116,7 @@
    /*
     * if not mother, request child.  Since that child only handles one
     * client at a time, it's safe to block as long as it takes.  Mother
-    * on the other hand can not block.
+    * on the other hand cannot block.
     */
    if (sendmsgn(s, &msg, 0, sockscf.state.type == PROC_MOTHER ? 0 : -1)
    != length) {
@@ -1806,8 +1806,8 @@
 
          /*
           * signals mother has set up but which we need ignore at this
-          * point, lest we accidentally run mothers signal handler if the
-          * child does not install it's own signal handler for the
+          * point, lest we accidentally run mother's signal handler if the
+          * child does not install its own signal handler for the
           * particular signal.
           * Later on, the child sets up its own signal handlers.
           */
@@ -1888,7 +1888,7 @@
 
 #if HAVE_PRIVILEGES
                if (sockscf.state.haveprivs) {
-                  /* don't need this privilege so permanently loose it. */
+                  /* don't need this privilege so permanently lose it. */
                   priv_delset(sockscf.privileges.privileged, PRIV_NET_PRIVADDR);
 
                   if (setppriv(PRIV_SET,
@@ -1952,13 +1952,13 @@
           * newprocinit() will close the old syslog descriptor, if any,
           * before opening a new one.  If we have started to use the
           * descriptor for something else already (e.g. due to dup(2)),
-          * newprocinit(), will still close the old descriptor, even
+          * newprocinit() will still close the old descriptor, even
           * though it's no longer a syslog descriptor.
           */
          newprocinit();
 
          /*
-          * This is minor optimization to make things faster for select(2)
+          * This is a minor optimization to make things faster for select(2)
           * by avoiding having two increasingly high-numbered descriptors
           * to check for, with most of the other descriptors in the lower-end.
           */
--- a/sockd/sockd_io.c
+++ b/sockd/sockd_io.c
@@ -77,9 +77,9 @@
 static int
 io_connectisinprogress(const sockd_io_t *io);
 /*
- * Returns true if "io" belongs to a connect whos current state is marked
- * as being in progress, but does not check whether it's status has changed
- * since last time state was updated.
+ * Returns true if "io" belongs to a connect whose current state is marked
+ * as being in progress, but does not check whether its status has changed
+ * since the last time state it was updated.
  * Returns false otherwise.
  */
 
@@ -195,7 +195,7 @@
  * "type", if not NULL, is filled in with the type of timeout that will
  * occur at that time, if any.
  *
- * Returns the number of seconds til the io object "io" will timeout.
+ * Returns the number of seconds till the io object "io" will timeout.
  *
  * 0 if the timeout has already been reached, or
  * -1 if no timeout on the io is currently set.
@@ -252,8 +252,8 @@
  * Note that this function must be called after the connect has completed,
  * as in the socks case (and some covenant cases) we need to send a
  * response back to the client before it will start sending us data.
- * We can thus not delay calling this function til we get ordinary i/o
- * from one side, as it's possible none will be coming til after we
+ * We can thus not delay calling this function till we get ordinary i/o
+ * from one side, as it's possible none will be coming till after we
  * have sent the response to the client.
  *
  * Returns 0 if the socket connected successfully.
@@ -311,7 +311,7 @@
  * if not 0, we have "overflowed" according to max bandwidth configured.
  * We can not attribute it to any given client though, so we penalize
  * all by delaying a little.  This object gives the earliest time at which we
- * can again do i/o over one of the object that has overflown it's bandwidth
+ * can again do i/o over one of the object that has overflown its bandwidth
  * limit.
  */
 static struct timeval bwoverflowtil;
@@ -361,7 +361,7 @@
          serr("%s: could not make rawsocket non-blocking", function);
 
 #if HAVE_PRIVILEGES
-   /* don't need this privilege any more, permanently loose it. */
+   /* don't need this privilege any more, permanently lose it. */
 
    if (sockscf.state.haveprivs) {
       priv_delset(sockscf.privileges.privileged, PRIV_NET_ICMPACCESS);
@@ -617,7 +617,7 @@
 #else /* !BAREFOOTD */
          /*
           * this process can continue independent of mother as long as it
-          * has clients, because each client has it's own unique
+          * has clients, because each client has its own unique
           * udp socket on the client-side also.
           */
 #endif /* !BAREFOOTD */
@@ -630,7 +630,7 @@
       }
 
       /*
-       * this needs to be after check of ack-pipe to limit error messages,
+       * this needs to be after the check of ack-pipe to limit error messages,
        * because the ack-pipe is a stream pipe, so hopefully we will handle
        * the EOF from mother on the ack-pipe before we get the error on
        * the data-pipe.
@@ -731,7 +731,7 @@
 
 #if BAREFOOTD
       /*
-       * Checked before, and checked again now, as it's status too may
+       * Checked before, and checked again now, as its status too may
        * change and it may become readable (again).
        */
       if (rawsocket != -1) {
@@ -745,8 +745,8 @@
        * We don't bother actually checking udp sockets for writability
        * because if the udp write ends up failing, it wouldn't make any
        * difference whether the socket was marked as writable or not; for
-       * all we know it's "writability" could have have been limited to a
-       * one byte write/packet, while the corresponding packet read was much
+       * all we know its "writability" could have have been limited to a
+       * single-byte write/packet, while the corresponding packet read was much
        * larger, in which case our write could have failed anyway.
        */
       FD_ZERO(udprset);
@@ -1157,7 +1157,7 @@
              * packet has been read.
              *
              * Try to read as many packets as we can as presumably there is
-             * a much greater risk of loosing packets on on the client side,
+             * a much greater risk of losing packets on on the client side,
              * since all the clients send to one address/socket.
              */
 
@@ -1823,7 +1823,7 @@
 #endif /* BAREFOOTD */
 
             /*
-             * Each client will have it's own target/dst object, set once
+             * Each client will have its own target/dst object, set once
              * we receive the first packet from it.  The client/source socket
              * is however the same for all clients.
              */
@@ -2314,7 +2314,7 @@
             /*
              * the client-socket is shared among many clients, so set it
              * regardless of bw-limits as we don't know from what
-             * client the packet is til we've read the packet.
+             * client the packet is till we've read the packet.
              *
              * XXX But what do we do if the bw overflows?  We can't know
              * that until we've read the packet and seen what client it's
--- a/sockd/sockd_negotiate.c
+++ b/sockd/sockd_negotiate.c
@@ -151,7 +151,7 @@
 static struct timeval *
 neg_gettimeout(struct timeval *timeout);
 /*
- * Fills in "timeout" with time til the first clients connection
+ * Fills in "timeout" with time till the first clients connection
  * expires.
  * Returns:
  *      If there is a timeout: pointer to filled in "timeout".
@@ -211,7 +211,7 @@
 static struct timeval *
 neg_gettimeout(struct timeval *timeout);
 /*
- * Fills in "timeout" with time til the first clients connection expires.
+ * Fills in "timeout" with time till the first clients connection expires.
  * Returns:
  *      If there is a timeout: pointer to filled in "timeout".
  *      If there is no timeout: NULL.
@@ -1237,8 +1237,8 @@
      * Normally there is no client data in Dante's case, but some clients
      * may piggy-back the payload together with the socks request, without
      * waiting for our response.  That is not legal to do, but some clients
-     * do it anyway, so we better support it.
-     * We therefor need to make sure we never read more of the payload than
+     * do it anyway, so we'd better support it.
+     * We therefore need to make sure we never read more of the payload than
      * we can send on to the i/o process, which will eventually need to
      * forward it to the destination.
      */
@@ -1319,7 +1319,7 @@
             if (permit) {
                /*
                 * Let the hostid-rule inherit settings from the client-rule,
-                * or use it's own.
+                * or use its own.
                 */
                if (rule_inheritoruse(&neg->crule,
                                      &cinfo,
@@ -1439,7 +1439,7 @@
       if (client.clientdatalen > 0) {
          slog(LOG_DEBUG,
               "%s: received client already has %lu bytes read from it.  "
-              "Must be a client that is changing it's http server target "
+              "Must be a client that is changing its http server target "
               "to %s, so request should already be parsed",
               function,
               (unsigned long)client.clientdatalen,
--- a/sockd/sockd_request.c
+++ b/sockd/sockd_request.c
@@ -1283,7 +1283,7 @@
              *
              * Unfortunately the meaning given in these standard provides
              * limited usability, so people "interpret" the standards more
-             * loose to get more practical functionality out of them.
+             * loosely to get more practical functionality out of them.
              *
              * - If the client provided an ip address when requesting the
              *   bind, we should only return remote connections matching
@@ -1398,7 +1398,7 @@
                               emsglen);
 
          /*
-          * XXX we should check whether it's possible receive any bindreply
+          * XXX we should check whether it's possible to receive any bindreply
           * also.  No need to stick around if no replies will be allowed.
           */
 
@@ -1467,10 +1467,10 @@
           * have a lot of complicated stuff for handling it in the i/o
           * process.
           *
-          * The procedure thus becomes that we use the clients TCP
+          * The procedure thus becomes that we use the client's TCP
           * address to evaluate what resources to allocate, and then
-          * use the clients UDP address to evaluate whether each individual
-          * packet should be passed through.  If the client sends us it's
+          * use the client's UDP address to evaluate whether each individual
+          * packet should be passed through.  If the client sends us its
           * UDP address in the udpassociate request, we can also evaluate
           * here whether it's at all possible that any udp packets from
           * it will be let through.  If not, we can just as well block it
@@ -1915,7 +1915,7 @@
 
                /*
                 * The problem is that both we and the process which receives
-                * the io packet needs to know when the client closes it's
+                * the io packet needs to know when the client closes its
                 * connection, but _we_ need to receive a query from the
                 * client on the connection as well, and the io process would
                 * get confused about that.  We try to hack around that
@@ -1944,7 +1944,7 @@
 
          if (rc == 0) {
             /*
-             * let client know what address we bound to on it's behalf.
+             * let client know what address we bound to on its behalf.
              */
 
             sockaddr2sockshost(&io.dst.laddr, &response.host);
@@ -1996,7 +1996,7 @@
          if (bindio.state.extension.bind) {
             sockshost2sockaddr(&bindio.dst.host, &bindio.dst.raddr);
 
-            /* won't create socket for this til we connect to the client. */
+            /* won't create socket for this till we connect to the client. */
             bzero(&bindio.dst.laddr, sizeof(bindio.dst.laddr));
             SET_SOCKADDR(&bindio.dst.laddr, AF_INET);
             TOIN(&bindio.dst.laddr)->sin_addr.s_addr = htonl(INADDR_ANY);
@@ -2014,7 +2014,7 @@
          sockaddr2sockshost(&bindio.src.laddr, &bindio.src.host);
 
          /*
-          * don't know what peer will be til we accept(2) it.
+          * don't know what peer will be till we accept(2) it.
           */
          bzero(&bindio.src.raddr, sizeof(bindio.src.raddr));
          SET_SOCKADDR(&bindio.src.raddr, bindio.src.laddr.ss_family);
@@ -2841,7 +2841,7 @@
           * Let the created socket be of the same address family as
           * the client tells us it will send packets from.  Note that
           * even if the client sends us an all-zero address now, even
-          * the all-zero address will have to have it's family
+          * the all-zero address will have to have its family
           * specified, as either ipv4 or ipv6.
           *
           * Exception is if the client does something so strange as to
@@ -2969,12 +2969,12 @@
 #else /* SOCKS_SERVER */
          /*
           * bind client-side address for receiving UDP packets, so we can tell
-          * the client where to send it's packets.
+          * the client where to send its packets.
           * XXX should perhaps have a global udprange option also, for clients
           *     that don't know what address they will be sending from.  This
-          *     rule is possibly not the one the client will finaly end up
+          *     rule is possibly not the one the client will finally end up
           *     using, in which the udprange specification would not apply,
-          *     but how can we know if the client doesn't tell us it's source
+          *     but how can we know if the client doesn't tell us its source
           *     address.
           * XXX add check for privileges on startup if range is privileged
           */
@@ -3740,7 +3740,7 @@
          /*
           * Possibly there is a route supporting an ipaddress destination,
           * even if there was no route supporting the hostname destination
-          * (e.g., there is only socks v4 route).  Therefor try resolving the
+          * (e.g., there is only a socks v4 route).  Therefore try to resolve the
           * destination locally before giving up on finding a route.
           *
           * We will need to resolve the destination sooner or later
--- a/sockd/sockd_udp.c
+++ b/sockd/sockd_udp.c
@@ -116,7 +116,7 @@
     *               maxsessions: 1 }
     *
     * Correspondingly, in Dante, the below will not work as the
-    * socks udp client can change it's destination at any time.
+    * socks udp client can change its destination at any time.
     *
     * pass { from: 10/8 to: 10.1.1.1/32 port = echo
     *        maxsessions: 10 }
--- a/lib/config_parse.c
+++ b/lib/config_parse.c
@@ -2628,7 +2628,7 @@
              * we are started with.  If logfiles created by that euid/egid are
              * not writable by our configured privileged userid (if any), it
              * means that upon SIGHUP we will be unable to re-open our own
-             * logfiles.  We therefor check whether the logfile(s) were created
+             * logfiles.  We therefore check whether the logfile(s) were created
              * by ourselves, and if so, make sure they have the right owner.
              */
             logtype_t *logv[] = { &sockscf.log, &sockscf.errlog };
--- a/lib/config_parse.y
+++ b/lib/config_parse.y
@@ -3174,7 +3174,7 @@
              * we are started with.  If logfiles created by that euid/egid are
              * not writable by our configured privileged userid (if any), it
              * means that upon SIGHUP we will be unable to re-open our own
-             * logfiles.  We therefor check whether the logfile(s) were created
+             * logfiles.  We therefore check whether the logfile(s) were created
              * by ourselves, and if so, make sure they have the right owner.
              */
             logtype_t *logv[] = { &sockscf.log, &sockscf.errlog };
--- a/lib/connectchild.c
+++ b/lib/connectchild.c
@@ -55,7 +55,7 @@
  * handles the socks negotiation and returns the proxy server's response back
  * to us.
  *
- * To avoid the client stepping on our (or rather our connect-childs) toes
+ * To avoid the client stepping on our (or rather our connect-child's) toes
  * while it negotiates with the proxy server, we temporarily let the
  * fd-index the client is using point at at dummy socket, while we use
  * the real socket to negotiate.  Then we set the clients fd to point back
--- a/lib/httpproxy.c
+++ b/lib/httpproxy.c
@@ -105,7 +105,7 @@
    }
 
    /*
-    * read til we get the eof response so there's no junk left in buffer
+    * read till we get the eof response so there's no junk left in buffer
     * for client, then return the response code.
     */
    eof = checked = readsofar = 0;
@@ -137,7 +137,7 @@
       readsofar += len;
 
       if ((strstr(buf, eofresponse_str)) == NULL)
-         continue; /* don't bother to start parsing til we've got it all. */
+         continue; /* don't bother to start parsing till we've got it all. */
       else
          eof = 1;
 
--- a/lib/iobuf.c
+++ b/lib/iobuf.c
@@ -154,7 +154,7 @@
     * In the client-case, we don't want to encode the packet on
     * every buffered write.  E.g. we don't want 100 putc(3)'s to
     * end up creating 100 gssapi-encoded one-byte packets.
-    * We therefore postpone encoding til we get a flush call, then
+    * We therefore postpone encoding till we get a flush call, then
     * encode the data we have, and write it as one token.
     * That means any already encoded data in the buffer must be written
     * before the unencoded data.
--- a/sockd/auth_pam.c
+++ b/sockd/auth_pam.c
@@ -122,7 +122,7 @@
     * some sort of busy-loop if we don't call pam_end(3) ever so
     * often.
     *
-    * Therefor, disregard all possible optimization stuff for now and
+    * Therefore, disregard all possible optimization stuff for now and
     * call pam_start(3) and pam_end(3) every time.
     */
 
--- a/sockd/method_uname.c
+++ b/sockd/method_uname.c
@@ -266,14 +266,14 @@
     * Very sadly we can't always do checking of the username/password here
     * since we don't know what authentication to use yet.  It could
     * be username, but it could also be PAM, or some future method.
-    * It depends on what the socks request is.  We therefor would have
+    * It depends on what the socks request is.  We therefore would have
     * liked to give the client success status back no matter what
     * the username/password is, and later deny the connection if need be
     * after we have gotten the request.
     *
     * That however creates problems with clients that, naturally, cache
     * the wrong username/password if they get success.
-    * We therefor check if we have a unique passworddb to use, and if so,
+    * We therefore check if we have a unique passworddb to use, and if so,
     * check the password here so we can return an immediate error to client.
     * This we can do because if the passworddb is unique there is
     * no chance of the result varying based to the client's request.
--- a/sockd/sockd.c
+++ b/sockd/sockd.c
@@ -199,7 +199,7 @@
    /*
     * The monitor-child is special, as there is only one and it
     * is shared/used by all processes, the mother processes
-    * included.  It therefor needs to be the first one created,
+    * included.  It therefore needs to be the first one created,
     * even before the mother processes.
     */
    if (childcheck(PROC_MONITOR) < 1) {
@@ -1594,9 +1594,9 @@
 
          if (child->type == PROC_IO) {
             /*
-             * don't receive anything back from i/o childs
-             * except the freeslot ack, as i/o childs are the
-             * last in the chain, so need to update this stat her.
+             * don't receive anything back from i/o children
+             * except the freeslot ack, as i/o children are the
+             * last in the chain, so need to update this stat here.
              */
             ++sockscf.stat.io.received;
 #if COVENANT
--- a/sockd/sockd_socket.c
+++ b/sockd/sockd_socket.c
@@ -156,7 +156,7 @@
 
       slog(LOG_DEBUG, "%s: re-bind(2) after unconnecting failed: %s.  "
                       "Current address is %s (was %s).  Trying to create a "
-                      "new socket instead, though we might loose some packets "
+                      "new socket instead, though we might lose some packets "
                       "doing so",
                       function,
                       strerror(errno),
--- a/lib/log.c
+++ b/lib/log.c
@@ -105,7 +105,7 @@
 dolog(const int priority, const char *buf,
       const size_t logprefixlen, const size_t messagelen);
 /*
- * Does the actual logging of the formated logmessage for slog()/vslog().
+ * Does the actual logging of the formatted logmessage for slog()/vslog().
  *
  * The last character in "buf", before the NUL, must be a newline.
  *
