Description: Fix the i386 !linux FTBFS because of socklen_t.
Bug-Debian: https://bugs.debian.org/845404
Forwarded: no
Author: Samuel Thibault <sthibault@debian.org>
Last-Update: 2017-01-09

--- a/osdep.m4
+++ b/osdep.m4
@@ -398,11 +398,14 @@
    for type in uint8_t "unsigned char"; do
        AC_TRY_COMPILE([
 #include <sys/types.h>
-#include <sys/socket.h>], [
+#include <sys/socket.h>
+#ifdef HAVE_STDINT_H
+#include <stdint.h>
+#endif], [
 struct sockaddr sa;
 $type *sa_len_ptr;
 sa_len_ptr = &sa.sa_len;
-if (((unsigned long long)sa_len_ptr) % 2 != 0)
+if (((uintptr_t)sa_len_ptr) % 2 != 0)
   return 0;],
        [AC_DEFINE_UNQUOTED(sa_len_type, [$type], [sa_len type])
         sa_len_type_found=t
